testsuite/classlibrary/TestPattern.sc

```diff
@@ -155,6 +155,21 @@ TestPattern : UnitTest {
 		this.assert(x.sum { |x| x.delta } == 4.5, "Psync with maxdur = quant should end no earlier than after dur");
 	}
 
+	test_Pfset_nilStream {
+		var x = 0, y = 0, oe = (), ie = (), clup = EventStreamCleanup.new;
+		oe = Pfset({ x = 1 }, p{}, { y = 2 }).asStream.next(ie);
+		this.assert(x == 1, "Pfset on nil stream should still call the initializer function");
+		this.assert(y == 2, "Pfset on nil stream should still call the cleanup function");
+		this.assert(oe.isNil, "Pfset on nil stream should return nil");
+		this.assert(ie.size.even, "Pfset on nil stream should add an even number of items to the input event");
+		// ie.size is 2 if Pfset adds 'addToCleanup' and 'removeFromCleanup' in sequence (presently);
```

Comment:
These 2 comment lines explain the line above (164) while the last 2 explain the line below; I'd move the first 2 lines up so they precede the code that they explain. (A new developer will read them as one block and it's not clear visually that they are separate concepts.)